### PR TITLE
Add an exported non-retriable error to throw during step functions

### PR DESCRIPTION
**Pull request by jpwilliams**
## Summary

Adds an exported `NonRetriableError` that can be used to throw during function runs to cease all further execution of a function.

```ts
inngest.createFunction("Example", "app/user.created", () => {
  try {
    // Do something
  } catch (cause) {
    throw new NonRetriableError("Something went wrong", { cause });
  }
});
```

## Related

- #45

### CODE REVIEW

**Review by djfarrelly — COMMENTED** [review]
I like this change @jpwilliams - Does this work for any step including the top level function handler? 

Our support for returning 4xx status codes (or any custom status codes) from the function handler is no longer working as is documented in our docs. Can we also support that or are we just going to go the custom error type? I'm fine either way, but we should get this out then update the docs ASAP.

**Comment by jpwilliams** [comment]
> I like this change @jpwilliams - Does this work for any step including the top level function handler?

Aha - only for the top level atm. We could also be able to trigger this within steps!

> Our support for returning 4xx status codes (or any custom status codes) from the function handler is no longer working as is documented in our docs. Can we also support that or are we just going to go the custom error type? I'm fine either way, but we should get this out then update the docs ASAP.

Detecting whether a user has returned `{ status: number; }` and then wrapping doesn't work too well, as we block users from returning an object with that shape. In addition, we're then giving access to a more internal detail; we should definitely provide first-class tooling for this rather than requiring the user to know a special number. 🪄

File: src/components/InngestCommHandler.ts [hunk 1]
```diff
@@ -439,9 +440,10 @@ export class InngestCommHandler<H extends Handler, TransformedRes> {
           runRes.data
         );
 
-        if (stepRes.status === 500) {
+        if (stepRes.status === 500 || stepRes.status === 400) {
```
[the remark this answers is not in the document]
Comment (jpwilliams) [reply src/components/InngestCommHandler.ts:443]: Aye great catch - I'll shift this to be able to throw during step execution too. :)

**Comment by tonyhb** [comment]
This is already in, right?

**Comment by jpwilliams** [comment]
> @tonyhb: This is already in, right?

Yes. :)

**Pull request closed without merging**